lib/TH/generic/THTensorMath.c

```diff
@@ -1066,6 +1066,30 @@ void THTensor_(cross)(THTensor *r_, THTensor *a, THTensor *b, int dimension)
                        r__data[2*r__stride] = a_data[0*a_stride]*b_data[1*b_stride] - a_data[1*a_stride]*b_data[0*b_stride];);
 }
 
+void THTensor_(cmax)(THTensor *r, THTensor *t, THTensor *src) {
+  THTensor_(resizeAs)(r, t);
+  TH_TENSOR_APPLY3(real, r, real, t, real, src,
```

Comment:
This is Stepanov's Mistake in action (it took Stepanov 15 years to get min and max right):

You get away with this in C because you are comparing a fixed set of types, but the logic is flawed here: Suppose `<` is only a strict weak order which has equivalent elements, and suppose that `a` and `b` are equivalent but non-equal. You really want `min(a, b)` and `max(a, b)` to be two different elements!

The correct way to write these algorithms is to keep the comparison consistent (and parcimonious, using only a single operator):

    max(a, b) = a < b ? b : a
    min(a, b) = a < b ? a : b

In a nutshell, all the ordering operations (min, max, stable partition, nth element, partial sort, etc.) need to be defined consistently with respect to one another.